src/sentry/integrations/source_code_management/commit_context.py

```diff
@@ -4,22 +4,43 @@
 from abc import ABC, abstractmethod
 from collections.abc import Mapping, Sequence
 from dataclasses import dataclass
-from datetime import datetime
+from datetime import datetime, timedelta, timezone
 from typing import Any
 
-from django.utils import timezone
+import sentry_sdk
+from django.utils import timezone as django_timezone
 
 from sentry import analytics
 from sentry.auth.exceptions import IdentityNotValid
 from sentry.integrations.models.repository_project_path_config import RepositoryProjectPathConfig
-from sentry.models.pullrequest import CommentType, PullRequestComment
+from sentry.locks import locks
+from sentry.models.commit import Commit
+from sentry.models.group import Group
+from sentry.models.groupowner import GroupOwner
+from sentry.models.options.organization_option import OrganizationOption
+from sentry.models.project import Project
+from sentry.models.pullrequest import (
+    CommentType,
+    PullRequest,
+    PullRequestComment,
+    PullRequestCommit,
+)
 from sentry.models.repository import Repository
 from sentry.users.models.identity import Identity
 from sentry.utils import metrics
+from sentry.utils.cache import cache
 
 logger = logging.getLogger(__name__)
 
 
+DEBOUNCE_PR_COMMENT_CACHE_KEY = lambda pullrequest_id: f"pr-comment-{pullrequest_id}"
+DEBOUNCE_PR_COMMENT_LOCK_KEY = lambda pullrequest_id: f"queue_comment_task:{pullrequest_id}"
```

Comment:
I see that this is taken from existing code, but is there a reason these are constant lambdas instead of just regularly defined functions?

I don't mind the functional idiom, but using `def` would have benefits including a type annotation on `pullrequest_id`.